src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,207 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private class Node<T> {
+
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.value = value;
+            this.prev = prev;
+            this.next = next;
+        }
+    }
+
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+    }
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            firstNodeAdd(value);
+        } else {
+            Node<T> addNode = new Node<>(tail, value, null);
+            tail.next = addNode;
+            tail = addNode;
+            size++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAdd(index);
+        if (size == 0) { //adding first element to empty LL
+            firstNodeAdd(value);
+            return;
+        }
+        Node<T> addNode = new Node<>(null, value, null);
+        Node<T> currentNode;
+        if (index == 0) { //adding first element to LL
+            currentNode = head;
+            currentNode.prev = addNode;
+            addNode.next = currentNode;
+            size++;
+            head = addNode;
+            return;
+        }
+        if (index <= size / 2) { //from head to tail
+            currentNode = head;
+            for (int i = 1; i < index; i++) {
+                currentNode = currentNode.next;
+            }
+        } else { //from tail to head
+            currentNode = tail;
+            for (int i = size - 1; i >= index; i--) {
+                currentNode = currentNode.prev;
+            }
+        }
+        addNode.prev = currentNode;
+        addNode.next = currentNode.next;
+        if (addNode.next == null) {
+            tail = addNode;
+        } else {
+            currentNode.next.prev = addNode;
+        }
+        currentNode.next = addNode;
+        size++;
```

Comment:
Create linkBefore() method.
```suggestion
            Node<T> target = getNode(index);
            linkBefore(value, target);
```

Reply:
not fixed